From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matti=20Lehtim=C3=A4ki?= <matti.lehtimaki@jolla.com>
Date: Fri, 23 Feb 2024 21:31:55 +0200
Subject: [PATCH] backport: tracker-miner-fs: Allow building without gstreamer

Build fails when using generic_media_extractor=libav and
not having gstreamer headers installed.
---
 src/miners/fs/tracker-main.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/miners/fs/tracker-main.c b/src/miners/fs/tracker-main.c
index 21f6166bc2e931714ed798747d3bd32cba23d88d..eb0b23883dfc8c0ab76424e3d9fc02e20a4c7d4e 100644
--- a/src/miners/fs/tracker-main.c
+++ b/src/miners/fs/tracker-main.c
@@ -35,7 +35,9 @@
 #include <glib-object.h>
 #include <glib/gi18n.h>
 
+#if defined(HAVE_GSTREAMER)
 #include <gst/gst.h>
+#endif
 
 #include <libtracker-miners-common/tracker-common.h>
 #include <libtracker-sparql/tracker-sparql.h>
@@ -989,7 +991,9 @@ main (gint argc, gchar *argv[])
 	/* Preempt possible registry updates, before tracker-extract-3 deals
 	 * with gstreamer plugins.
 	 */
+#if defined(HAVE_GSTREAMER)
 	gst_init (NULL, NULL);
+#endif
 
 	/* Translators: this messagge will apper immediately after the
 	 * usage string - Usage: COMMAND <THIS_MESSAGE>
